streams/src/test/java/org/apache/kafka/streams/state/internals/CachingKeyValueStoreTest.java

```diff
@@ -121,21 +122,71 @@ public void shouldAvoidFlushingDeletionsWithoutDirtyKeys() {
     }
 
     @Test
-    public void shouldCloseAfterErrorWithFlush() {
+    public void shouldCloseWrappedStoreAfterErrorDuringCacheFlush() {
+        setUpCloseTests();
+        cache.flush(CACHE_NAMESPACE);
+        EasyMock.expectLastCall().andThrow(new NullPointerException("Simulating an error on flush"));
+        EasyMock.replay(cache);
+        EasyMock.reset(underlyingStore);
+        underlyingStore.close();
+        EasyMock.replay(underlyingStore);
+
+        try {
+            store.close();
+        } catch (final RuntimeException exception) {
+            EasyMock.verify(underlyingStore);
+        }
+    }
+
+    @Test
+    public void shouldCloseWrappedStoreAfterErrorDuringCacheClose() {
+        setUpCloseTests();
+        cache.flush(CACHE_NAMESPACE);
+        cache.close(CACHE_NAMESPACE);
+        EasyMock.expectLastCall().andThrow(new NullPointerException("Simulating an error on close"));
+        EasyMock.replay(cache);
+        EasyMock.reset(underlyingStore);
+        underlyingStore.close();
+        EasyMock.replay(underlyingStore);
+
         try {
-            cache = EasyMock.niceMock(ThreadCache.class);
-            context = new InternalMockProcessorContext(null, null, null, null, cache);
-            context.setRecordContext(new ProcessorRecordContext(10, 0, 0, topic, null));
-            store.init(context, null);
-            cache.flush("0_0-store");
-            EasyMock.expectLastCall().andThrow(new NullPointerException("Simulating an error on flush"));
-            EasyMock.replay(cache);
             store.close();
-        } catch (final NullPointerException npe) {
-            assertFalse(underlyingStore.isOpen());
+        } catch (final RuntimeException exception) {
+            EasyMock.verify(underlyingStore);
         }
     }
 
+    @Test
+    public void shouldCloseCacheAfterErrorDuringStateStoreClose() {
```

Comment:
What do we actually test here? Closing the underlying store is the last thing we do, so it does not affect the happy path of flushing/closing the cache (what we verify at the end). Not sure what the value of this test is?

Reply:
I see what you mean. IMO we should test also this, because tests are also here to make refactoring safe. Let's assume somebody changes the order of the closes, we would still have a check.